components/datasets/form/steps/Step1.js

```diff
@@ -57,6 +64,42 @@ class Step1 extends React.Component {
     this.props.onChange({ legend });
   }
 
+  onSubscribableChange(obj) {
+    const { subscribable } = this.props.form;
+    const newSubscribable = subscribable.slice(0);
+    if (obj.type) {
+      newSubscribable[obj.id].type = obj.type;
+    } else if(obj.value) {
+      newSubscribable[obj.id].value = obj.value;
+    }
+    this.props.onChange({ subscribable: newSubscribable });
+  }
+
+  onSubscribableCheckboxChange(checked) {
+    this.setState({
+      subscribableSelected: checked,
+    });
+    if (checked) {
+      this.props.onChange({ subscribable: [{ type: '', value:'', id: 0 }] });
+    } else {
+      this.props.onChange({ subscribable: [] });
+    }
+  }
+
+  handleRemoveSubscription(index) {
+    const { subscribable } = this.props.form;
+    const newSubscribable = subscribable.slice(0);
+    newSubscribable.splice(index, 1)
+    this.props.onChange({ subscribable: newSubscribable });
+  }
+
+  handleAddSubscription() {
+    const { subscribable } = this.props.form;
+    const newSubscribable = subscribable.slice(0);
+    newSubscribable.push({ type: '', value: '', id: subscribable.length })
```

Comment:
What happens if I create an element, then create another one but I decide to remove the first one. Then try to create another one, the would have the same id... => ERROR. Use Date.now() to be sure that they will be different

Reply:
```
  handleAddSubscription() {
    const { subscribable } = this.props.form.subscribable.slice(0);
    subscribable.push({ type: '', value: '', id: Date.now() });
    this.props.onChange({ subscribable });
  }

```

Reply:
I've used a counter better than Date.now(), WDYT?
``` this.subscribableCounter``` that I'm initializing in the constructor and updating everytime the add button is clicked

Reply:
I don't like it. What would happen if I edit one subscribable dataset that already exists? Would the counter be initialized at 0? What if there is already a 0 id? You need to be sure that the id is new and unique. Time is unique and I think that's better than add a library or something else